src/GrumPHP/Parser/Php/PhpParser.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace GrumPHP\Parser\Php;
+
+use GrumPHP\Collection\NodeVisitorsCollection;
+use GrumPHP\Collection\ParseErrorsCollection;
+use GrumPHP\Parser\ParserInterface;
+use PhpParser\Parser;
+use PhpParser\Lexer\Emulative;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\NameResolver;
+use PhpParser\NodeVisitor;
+use PhpParser\Error;
+use SplFileInfo;
+
+/**
+ * Class PhpParser
+ *
+ * @package GrumPHP\Parser\Php
+ */
+class PhpParser implements ParserInterface
+{
+    /**
+     * @var NodeVisitorsCollection|NodeVisitor[]
+     */
+    protected $nodeVisitors;
+
+    public function __construct()
+    {
+        $this->nodeVisitors = new NodeVisitorsCollection();
+    }
+
+    /**
+     * @param NodeVisitor $visitor
+     * @param array       $options
+     */
+    public function addNodeVisitor(NodeVisitor $visitor)
+    {
+        if ($this->nodeVisitors->contains($visitor)) {
+            return;
+        }
+
+        $this->nodeVisitors->add($visitor);
+    }
+
+    /**
+     * @return NodeVisitorsCollection|NodeVisitor[]
+     */
+    public function getNodeVisitors()
+    {
+        return $this->nodeVisitors;
+    }
+
+    /**
+     * @param SplFileInfo $file
+     *
+     * @return GrumPHP\Collection\ParseErrorsCollection
+     */
+    public function parse(SplFileInfo $file)
+    {
+        $filename  = $file->getRealPath();
+
+        $errors    = new ParseErrorsCollection();
+
+        $parser    = new Parser(new Emulative);
```

Comment:
Maybe the parser initialisation should be done by a factory in the service container? This will make it easier to test.